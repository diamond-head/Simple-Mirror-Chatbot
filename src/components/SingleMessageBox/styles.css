.single-msg-box__container {
}

.single-msg-box__wrapper {
  border-radius: 12px;
  border: 0.5px solid var(--text-gray-50);
  padding: 16px;
  word-break: break-word;
  word-wrap: break-word;
  white-space: pre-wrap;
  text-align: left;
  width: auto;
  max-width: 100%;
  min-width: 0;
}

.single-msg-box__wrapper.left {
  background-color: var(--white);
  color: var(--text-gray);
}

.single-msg-box__wrapper.right{
  background-color: var(--primary-color);
  color: var(--white);
}