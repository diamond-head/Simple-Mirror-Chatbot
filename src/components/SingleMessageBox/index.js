import React from 'react';

import "./styles.css";

export default function SingleMessageBox({ content, side }) {
  return (
    <div className="single-msg-box__container">
      <div className={`single-msg-box__wrapper ${side}`}>
        {content}
      </div>
    </div>
  );
}