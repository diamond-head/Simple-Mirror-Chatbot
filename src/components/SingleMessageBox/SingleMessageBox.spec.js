import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SingleMessageBox from './index';

describe("<SingleMessageBox /> component", () => {
  it("SingleMessageBox snapshot renders", () => {
    const component = renderer.create(
      <SingleMessageBox content={'Hi'} side="left" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render an single-message-box left side content", () => {
    const wrapper = shallow(<SingleMessageBox content={'Hi'} side="left" />);
    expect(wrapper.find('.single-msg-box__wrapper.left').text()).toBe("Hi");
  });

  it("render an single-message-box right side content", () => {
    const wrapper = shallow(<SingleMessageBox content={'Hi'} side="right" />);
    expect(wrapper.find('.single-msg-box__wrapper.right').text()).toBe("Hi");
  });
});
