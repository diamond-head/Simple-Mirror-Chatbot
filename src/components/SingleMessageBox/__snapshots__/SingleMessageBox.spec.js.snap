// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleMessageBox /> component SingleMessageBox snapshot renders 1`] = `
<div
  className="single-msg-box__container"
>
  <div
    className="single-msg-box__wrapper left"
  >
    Hi
  </div>
</div>
`;
